### ISSUE TITLE
Oj.mimic_JSON HTML escaping

### ISSUE THREAD

**Issue by robertasg** [ohler55/oj#517]
`.mimic_JSON` seems to introduce HTML escaping that's not overridable, as it changes the escape mode to `JXEsc`.

It doesn't seem to behave like the `JSON` gem. Here's some output from `irb`.

```
2.5.1 :001 > require 'json'
 => true 
2.5.1 :002 > require 'oj'
 => true 
2.5.1 :003 > JSON.dump('<>' => '<>')
 => "{\"<>\":\"<>\"}" 
2.5.1 :004 > Oj.mimic_JSON
 => JSON 
2.5.1 :005 > JSON.dump('<>' => '<>')
 => "{\"\\u003c\\u003e\":\"\\u003c\\u003e\"}" 
```

**Comment by ohler55** [comment]
Hmm, you are correct. That mode was made explicitly for the JSON gem. I don't know if the gem changed or when I tried it out it was under Rails and that had an impact. I'll do some more testing and see what it should be.

**Comment by ohler55** [comment]
Please try the `escape` branch.

**Comment by robertasg** [comment]
@ohler55 Neat! Seems to work alright. I've added this test to test the change.

```
diff --git a/test/json_gem/json_generator_test.rb b/test/json_gem/json_generator_test.rb
index f71a1bf..a5a1616 100644
--- a/test/json_gem/json_generator_test.rb
+++ b/test/json_gem/json_generator_test.rb
@@ -54,6 +54,7 @@ EOT
     parsed_json = JSON.parse(json)
     assert_equal({"1"=>2}, parsed_json)
     assert_equal '666', JSON.generate(666)
+    assert_equal("{\"<>\":\"<>\"}", JSON.generate("<>" => "<>"))
   end
 
   def test_generate_pretty
```

**Comment by ohler55** [comment]
Can't really add the test there. Those tests are taken directly from the JSON gem test suite. Adding or changing them makes the compatibility tests something different than the gem standard. Adding the test to `test_compat.rb` is the place to put it. I noticed the tests failed so I'll have to create a new escape table for the JSON gem.

**Comment by robertasg** [comment]
We can definitely do this then:

```
diff --git a/test/test_compat.rb b/test/test_compat.rb
index bdb6cf1..1d8c137 100755
--- a/test/test_compat.rb
+++ b/test/test_compat.rb
@@ -243,6 +243,11 @@ class CompatJuice < Minitest::Test
     assert_equal({"a\nb" => true, "c\td" => false}, obj)
   end
 
+  def test_hash_escaping
+    json = Oj.to_json('<>' => '<>')
+    assert_equal(json, '{"<>":"<>"}')
+  end
+
   def test_bignum_object
     dump_and_load(7 ** 55, false)
   end
```

Is there any way I can help to move this forward?

**Comment by ohler55** [comment]
I think you just did. :-)

**Comment by ohler55** [comment]
Released 3.7.5.

**Comment by robertasg** [comment]
Woah. Amazing. Thanks!
